jwst/lib/set_telescope_pointing.py

```diff
@@ -992,4 +1104,45 @@ def _add_axis_3(model):
     wcsaxes = model.meta.wcsinfo.wcsaxes
     if wcsaxes is not None and wcsaxes == 3:
         model.meta.wcsinfo.ctype3 = "WAVE"
-        model.meta.wcsinfo.cunit3 = "um"
\ No newline at end of file
+        model.meta.wcsinfo.cunit3 = "um"
+
+
+def calc_rotation_matrix(angle, vparity=1):
+    """ Calculate the rotation matrix
+
+    Parameters
+    ----------
+    angle: float in radians
+        The angle to create the matrix
+
+    vparity: {1, -1}
+        The x-axis parity, usually taken from
+        the JWST SIAF parameter VIdlParity
+
+    Returns
+    -------
+    matrix: [cd1_1, cd1_2, cd2_1, cd2_2]
+        The rotation matrix
+
+    Notes
+    -----
+    The rotation is
+
+       ----------------
+       | cd1_1  cd2_1 |
+       | cd1_2  cd2_2 |
+       ----------------
+
+    where:
+        cd1_1 = vparity * cos(angle)
+        cd1_2 = sin(angle)
+        cd2_1 = -1 * vparity * sin(angle)
+        cd2_2 = cos(angle)
+    """
+
+    cd1_1 = vparity * cos(angle)
+    cd1_2 = sin(angle)
+    cd2_1 = -1 * vparity * sin(angle)
+    cd2_2 = cos(angle)
```

Comment:
BTW, it's somewhat faster to use `vparity * -cos(angle)`.
It's not part of this PR but since we are looking at these functions now,  it's ~10x faster to use `math.cos` instead of `np.cos`. Since there's no reason here to use the `numpy` trig functions we should switch to using `math`.